### PR TITLE
Fix for enabling the certificate store to work on cloud services.

### PR DESCRIPTION
**Pull request by jasho**
Fix for the RSACryptoServiceProvider to use Machine key store to enable the library to run in an Azure App Service.

### CODE REVIEW

File: src/Mews.Eet/Dto/Certificate.cs [hunk 1]
```diff
@@ -11,8 +11,8 @@ public Certificate(string password, byte[] data)
             Password = password;
             Data = data;
             Key = ComputeKey();
-            X509Certificate2 = new X509Certificate2(Data, Password);
-        }
+            X509Certificate2 = new X509Certificate2(Data, Password, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet | X509KeyStorageFlags.PersistKeySet);
+    }
```
Comment (jirihelmich) [inline src/Mews.Eet/Dto/Certificate.cs]: Revert the indent.

Comment (jasho) [inline src/Mews.Eet/Dto/Certificate.cs]: Fixed.

**Comment by jirihelmich** [comment]
It seems to cause different issues on Azure when loading files from Blob Storage now. Any ideas how to fix that?

```
{
   "Type":"System.Security.Cryptography.CryptographicException",
   "Message":"Keyset does not exist\r\n",
   "InnerExceptions":[

   ],
   "StackTrace":[
      "   at System.Security.Cryptography.Utils.CreateProvHandle(CspParameters parameters, Boolean randomKeyContainer)",
      "   at System.Security.Cryptography.Utils.GetKeyPairHelper(CspAlgorithmType keyType, CspParameters parameters, Boolean randomKeyContainer, Int32 dwKeySize, SafeProvHandle& safeProvHandle, SafeKeyHandle& safeKeyHandle)",
      "   at System.Security.Cryptography.RSACryptoServiceProvider.GetKeyPair()",
      "   at System.Security.Cryptography.RSACryptoServiceProvider..ctor(Int32 dwKeySize, CspParameters parameters, Boolean useDefaultKeySize)",
      "   at System.Security.Cryptography.X509Certificates.X509Certificate2.get_PrivateKey()",
      "   at Mews.Eet.Communication.SoapSigner.GetRsaKey(SignAlgorithm signAlgorithm, X509Certificate2 certificate)",
      "   at Mews.Eet.Communication.SoapSigner..ctor(X509Certificate2 certificate, SignAlgorithm signAlgorithm)",
      "   at Mews.Eet.Communication.SoapMessage.GetSignedXmlDocument(X509Certificate2 certificate, SignAlgorithm signAlgorithm)",
      "   at Mews.Eet.Communication.SoapClient.<SendAsync>d__22`2.MoveNext()",
      "--- End of stack trace from previous location where exception was thrown ---",
      "   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)",
      "   at Mews.Eet.Communication.EetSoapClient.<SendRevenueAsync>d__17.MoveNext()",
      "--- End of stack trace from previous location where exception was thrown ---",
      "   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)",
      "   at Mews.Eet.EetClient.<SendRevenueAsync>d__13.MoveNext()",
      "--- End of stack trace from previous location where exception was thrown ---",
      "   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)",
      "   at ..."
   ]
}
```

File: src/Mews.Eet/Communication/SoapSigner.cs [hunk 2]
```diff
@@ -96,6 +96,7 @@ private RSACryptoServiceProvider GetRsaKey(SignAlgorithm signAlgorithm, X509Cert
                 var key = certificate.PrivateKey as RSACryptoServiceProvider;
                 var enhCsp = new RSACryptoServiceProvider().CspKeyContainerInfo;
                 var cspparams = new CspParameters(enhCsp.ProviderType, enhCsp.ProviderName, key.CspKeyContainerInfo.KeyContainerName);
```
Comment (siroky) [inline src/Mews.Eet/Communication/SoapSigner.cs]: cspParameters and maybe name the enhCsp better too

**Pull request merged**